### PR TITLE
goreleaser for semantic versioning and docker image releases

### PR DESCRIPTION
**Pull request by sridhav**
Using goreleaser for semantic versioning and docker release process.

### CODE REVIEW

**Comment by sridhav** [comment]
@nurland can you review this PR and let me know what do you think? It's been tested on my fork, I can run few more tests once I get an approval about the process am using. Thanks!

FYI @sana-jawad @mallow111

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +1,26 @@
 language: go
 
+branches:
+  except:
+    - master
+  only:
+    - release
 go:
   - 1.13.x
```
Comment (mallow111) [inline .travis.yml:9]: should we update to 1.14.x version, which is the latest.

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
 
 services:
   - docker
 
 script:
-  - make
+  - go test ./...
+  - curl -sfL https://git.io/goreleaser | sh -s -- check
```
Comment (mallow111) [inline .travis.yml:16]: curl -sfL https://git.io/goreleaser is use in the below as well, do you wan to define a variable for this and used everywhere?

File: .travis.yml [hunk 3]
```diff
@@ -1,19 +1,26 @@
 language: go
 
+branches:
+  except:
+    - master
+  only:
+    - release
```
Comment (nurland) [inline .travis.yml:7]: why do we need release branch, can we use master?

File: .travis.yml [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
   - docker
 
 script:
-  - make
+  - go test ./...
+  - curl -sfL https://git.io/goreleaser | sh -s -- check
 
 before_deploy:
-  - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+  - test -n "$TRAVIS_TAG" && docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
```
Comment (nurland) [inline .travis.yml:19]: @sridhav  can we use existing way to pass docker password instead? Less chances to expose it in the logs

File: Dockerfile [hunk 5]
```diff
@@ -1,20 +1,12 @@
...
-
-COPY pkg ./pkg
-
-RUN curl -o /sloop/aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.14.6/2019-08-22/bin/linux/amd64/aws-iam-authenticator \
+FROM alpine as build
+RUN apk add --no-cache curl
+RUN curl -o /aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.14.6/2019-08-22/bin/linux/amd64/aws-iam-authenticator \
   && wait \
-  && chmod +x /sloop/aws-iam-authenticator
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s" -installsuffix cgo -o sloop ./pkg/sloop
```
Comment (nurland) [inline Dockerfile:13]: what will be the process of building image locally?

File: .goreleaser.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    image_templates:
+      - "salesforce/sloop:latest"
+      - "salesforce/sloop:{{ .Tag }}"
+    dockerfile: Dockerfile
+    build_flag_templates:
+      - "--pull"
+      - "--label=org.opencontainers.image.created={{.Date}}"
+      - "--label=org.opencontainers.image.name={{.ProjectName}}-captain"
+      - "--label=org.opencontainers.image.revision={{.FullCommit}}"
+      - "--label=org.opencontainers.image.version={{.Version}}"
```
Comment (nurland) [inline .goreleaser.yml:32]: how these vars are populated? where do we get them from? e.g. {{.Version}} and {{.Tag}}

File: .travis.yml (hunk 2 above)
Comment (sridhav) [reply .travis.yml:16]: @mallow111 actually i should remove this line, it was just there for a validation. We dont need this line

File: .goreleaser.yml (hunk 6 above)
Comment (sridhav) [reply .goreleaser.yml:32]: @nurland goreleaser inherently has these variables -- https://goreleaser.com/customization/templates/

File: .travis.yml (hunk 4 above)
Comment (sridhav) [reply .travis.yml:19]: Travis allows you to mask any environment variables.

File: .travis.yml (hunk 1 above)
Comment (sridhav) [reply .travis.yml:9]: @mallow111 not sure if the source is tested with  golang 1.14.x, hence just use 1.13.x

**Comment by sridhav** [comment]
@nurland I will open a new PR for these changes. I have two different emails in the commits, which is throwing this cla issue. 

FYI @mallow111 @sana-jawad

**Pull request closed without merging**